cmake/OpenCVFindIPP.cmake

```diff
@@ -151,7 +151,7 @@ macro(ipp_detect_version)
         if("${name}" STREQUAL "core")  # https://github.com/opencv/opencv/pull/19681
           if(OPENCV_FORCE_IPP_EXCLUDE_LIBS OR OPENCV_FORCE_IPP_EXCLUDE_LIBS_CORE
               OR (UNIX AND NOT ANDROID AND NOT APPLE
-                  AND (CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+                  AND CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang|Intel"
```

Comment:
AFAIK, oneAPI compiler is identified as "Clang" compiler (at least for 2021.1).

```
CMAKE_CXX_COMPILER=/opt/intel/oneapi/compiler/2021.1.1/linux/bin/clang++
CMAKE_CXX_COMPILER=/opt/intel/oneapi/compiler/2021.1.1/linux/bin/clang++
CMAKE_CXX_COMPILER_ABI=ELF
CMAKE_CXX_COMPILER_AR=/opt/intel/oneapi/compiler/2021.1.1/linux/bin/llvm-ar
CMAKE_CXX_COMPILER_AR=/opt/intel/oneapi/compiler/2021.1.1/linux/bin/llvm-ar
CMAKE_CXX_COMPILER_ARCHITECTURE_ID=
CMAKE_CXX_COMPILER_ARG1=
CMAKE_CXX_COMPILER_ENV_VAR=CXX
CMAKE_CXX_COMPILER_FRONTEND_VARIANT=GNU
CMAKE_CXX_COMPILER_ID=Clang                            <<<====== here
```

Intel Compiler (from "Intel(R) Parallel Studio XE" product) is a different product.

Reply:
There are two compilers in oneAPI, I used `icc` which identifies as "Intel" (I think it is called _classic_ in terms of oneAPI package):
```
--     C++ Compiler:                /opt/intel/oneapi/compiler/2021.2.0/linux/bin/intel64/icc  (ver 20.2.2.20210228)
```